portfolio/src/test/java/com/google/sps/NameTest.java

```diff
@@ -0,0 +1,47 @@
+package com.google.sps.servlets;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletOutputStream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnit4.class)
+public final class NameTest extends Mockito {
+  private final DataServlet servlet = new DataServlet();
+
+  @Mock
+  private HttpServletRequest request;
+  @Mock
+  private HttpServletResponse response;
+  @Mock
+  private ServletConfig servletConfig;
+  @Mock
+  private ServletOutputStream outputStream;
+
+  @Test
+  public void getMyName() throws IOException {
+   when(servlet.getServletConfig()).thenReturn(servletConfig);
+   when(response.getOutputStream()).thenReturn(outputStream);
```

Comment:
hmmm in your DataServlet code you are invoking response.getWriter() not response.getOutputStream(), so if you want to keep this call as is, you will need to change your servlet code. Also see [this](https://stackoverflow.com/questions/4746113/servlets-in-java-both-getwriter-and-getoutputstream#:~:text=To%20prevent%20this%20mix%2Dup,Writer%20for%20writing%20textual%20content.&text=Generally%20this%20is%20because%20getOutputStream,used%20to%20write%20textual%20content.) resource for using getWriter() vs. getOutputStream(). If you end up using getWriter, you should use the StringWriter/PrintWriter textual buffers as opposed to outputstream

Reply:
that makes more sense. ill make those changes. it seems like outputstream reads differently than stringwriter, and it would make 100x more sense to use stringwriter, haha.